MaxText/optimizers.py

```diff
@@ -0,0 +1,120 @@
+"""
```

Comment:
@ZhiyuLi-goog I haven't seen justification for this implemetnation.

Reply:
Discussed offline.
> `optax adam` is not the same as `adam_pax` and I got verified in [this doc](https://colab.corp.google.com/drive/1lB5jdqQni0qZsFKBhUYRbFEJGytHhxe3).
I went through the code, updates of step and momentum are different in orders. I got convergence regression in using optax adam.